src/raglite/_insert.py

```diff
@@ -46,7 +46,10 @@ def _create_chunk_records(
             )
     else:
         # Embed the full chunks, including the current Markdown headings.
-        full_chunk_embeddings = embed_sentences([str(chunk) for chunk in chunks], config=config)
+        full_chunk_embeddings = embed_text(
+            [str(chunk_record.content) for chunk_record in chunk_records], config=config
+        )
+
```

Comment:
I think this change doesn't actually do anything: `str(chunk_record)` should be equivalent to `chunk.content` (snippet from `_database.py:class Chunk`):
```python
    @property
    def content(self) -> str:
        """Return this chunk's contextual heading and body."""
        return f"{self.headings.strip()}\n\n{self.body.strip()}".strip()

    def __str__(self) -> str:
        """Return this chunk's content."""
        return self.content
```

```suggestion
        full_chunk_embeddings = embed_sentences([str(chunk) for chunk in chunks], config=config)
```

Reply:
Oh, good catch! Was too quick with this comment.

Reply:
but no need to recast to str, i will just put chunk_record.content